src/Symfony/Component/Security/Core/Encoder/SodiumPasswordEncoder.php

```diff
@@ -99,4 +99,20 @@ public function isPasswordValid($encoded, $raw, $salt)
 
         throw new LogicException('Libsodium is not available. You should either install the sodium extension, upgrade to PHP 7.2+ or use a different encoder.');
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function needsRehash(string $encoded): bool
```

Comment:
Should we add a check for BCrypt passwords too?

Reply:
It's deprecated so I don't think we should improve it further.